packages/firebase_remote_config/firebase_remote_config_platform_interface/lib/src/method_channel/method_channel_firebase_remote_config.dart

```diff
@@ -224,7 +234,14 @@ class MethodChannelFirebaseRemoteConfig extends FirebaseRemoteConfigPlatform {
     try {
       await channel.invokeMethod('RemoteConfig#setDefaults', <String, dynamic>{
         'appName': app.name,
-        'defaults': defaultParameters
+        'defaults': Map<String, dynamic>.from(
+          defaultParameters.map(
+            (key, value) => MapEntry(
+              key,
+              _isSupportedType(value) ? value : json.encode(value),
```

Comment:
I don't think that should be the responsibility of `setDefaults` to encode a value. I would expect the function to be called with an already serialized parameter

Reply:
`Map<String, dynamic>` type of the argument suggests you can pass anything inside (which happened in #2074).
Native APIs are not symmetric, iOS [does have json-related helper](https://firebase.google.com/docs/reference/swift/firebaseremoteconfig/api/reference/Classes/RemoteConfigValue#jsonvalue), android doesn't. Since we can't explicitly forbid non-primitive types on type level, I think having this helper method would be useful. An alternative would be asserting incoming value types and throwing on anything other than bool, number, or string, but I don't see this behavior as developer-friendly, especially taking into account we can provide a convenient method.

Agree, that encoding should be done not on a method channel level, so moving to `packages/firebase_remote_config/firebase_remote_config/lib/src/remote_config.dart`

Reply:
> Since we can't explicitly forbid non-primitive types on type level, I think having this helper method would be useful

I disagree. Instead I think throwing is the way to go.

The fact that Dart type system does not allows us to statically prevent values other than primitive doesn't mean that we should handle non-primitive values.

Map<String, dynamic> is the Dart standard for JSON structures allowing only primitives.
Most function taking such Map as parameter consider non-primitive values as an error.


Using "encode" here is non-standard and is trying to patch mistakes from the users.

Reply:
> The fact that Dart type system does not allows us to statically prevent values other than primitive doesn't mean that we should handle non-primitive values.

dart not allowing us to prevent values other than primitives shouldn't limit us in making the API more developer-friendly either 😉  

> Using "encode" here is non-standard and is trying to patch mistakes from the users.

I don't see it as a patch for a mistake. A developer wanting to pass a `Map` as a config value is hardly a mistake (as well as willing to read that `Map` back via `asJson`)